packages/headless/src/auth/saml-provider.test.ts

```diff
@@ -0,0 +1,149 @@
+import {IsomorphicHistory} from './isomorphic-history';
+import {IsomorphicLocation} from './isomorphic-location';
+import {
+  buildSamlProvider,
+  SamlProvider,
+  SamlProviderOptions,
+} from './saml-provider';
+
+describe('buildSamlProvider', () => {
+  const handshakeToken = 'token';
+  let options: Required<SamlProviderOptions>;
+  let request: jest.Mock<any, any>;
+  let provider: SamlProvider;
+
+  function initSamlProvider() {
+    provider = buildSamlProvider(options);
+  }
+
+  function buildMockLocation(): IsomorphicLocation {
+    return {hash: '', href: ''};
+  }
+
+  function buildMockHistory(): IsomorphicHistory {
+    return {replaceState: jest.fn()};
+  }
+
+  function assertHandshakeTokenSent() {
+    expect(request).toHaveBeenCalledWith(
+      expect.any(String),
+      expect.objectContaining({body: JSON.stringify({handshakeToken})})
+    );
+  }
+
+  beforeEach(() => {
+    request = jest.fn();
+
+    options = {
+      organizationId: '',
+      provider: '',
+      location: buildMockLocation(),
+      history: buildMockHistory(),
+      request,
+    };
+
+    initSamlProvider();
+  });
+
+  describe('#login', () => {
+    // TODO: test POST to avoid writing to url.
+    // TODO: url environments
```

Comment:
I will allow configuring the url to support different environments and regions. ([slack thread](https://coveo.slack.com/archives/C029XM38080/p1637588952019200?thread_ts=1637352288.019100&cid=C029XM38080))